controller/boot/bootmultiway/multiway.py

```diff
@@ -21,18 +14,14 @@
 CHANGE = Pin(5, Pin.OUT)
 
 
-class MultiwaySwitch():
+class MultiwaySwitch(MQTTDevice):
 
     def __init__(self):
-        self.client = MQTTClient(MQTT_CLIENT_ID_RANDOM, MQTT_BROKER_ADDRESS, port=MQTT_PORT, keepalive=KEEP_ALIVE_TIME_SEC)
-        self.next_scheduled_ping_time = 0
         self.previous_state = -1
-        self._client_setup()
-    
-    def _client_setup(self):
-        self.client.set_last_will(TOPIC_LAMP_GET_ONLINE, FALSE_STRING)
-        self.client.connect()
-        self.client.publish(TOPIC_LAMP_GET_ONLINE, TRUE_STRING)
+        super().__init__(TOPIC_LAMP_GET_ONLINE)
+
+
+    def setup_subscriptions(self):
         self.client.set_callback(self._siri_switch_handler)
         self.client.subscribe(TOPIC_LAMP_SET_ON)
         self._physical_switch_handler()
```

Comment:
I think it would make sense to either rename setup_subscriptions() in MQTTDevice to setup() or to introduce a new method "finalize_setup()" in MQTTDevice because the call  _physical_switch_handler() has nothing to do with subscriptions. But has to be called after the initial setup.